recipes-extended/xen/files/libxl-retry-qmp-pci-add.patch

```diff
@@ -0,0 +1,25 @@
+--- a/tools/libxl/libxl_qmp.c
++++ b/tools/libxl/libxl_qmp.c
+@@ -876,6 +876,7 @@ int libxl__qmp_pci_add(libxl__gc *gc, in
+     libxl__json_object *args = NULL;
+     char *hostaddr = NULL;
+     int rc = 0;
++    int i = 0;
+ 
+     qmp = libxl__qmp_initialize(gc, domid);
+     if (!qmp)
+@@ -907,8 +908,12 @@ int libxl__qmp_pci_add(libxl__gc *gc, in
+     if (pcidev->permissive)
+         qmp_parameters_add_bool(gc, &args, "permissive", true);
+ 
+-    rc = qmp_synchronous_send(qmp, "device_add", args,
+-                              NULL, NULL, qmp->timeout);
++    do {
++        rc = qmp_synchronous_send(qmp, "device_add", args,
++                                  NULL, NULL, qmp->timeout);
++        sleep(1);
```

Comment:
This will always sleep.
```
if (rc)
    sleep(1);
```
To avoid it on success.

Have you measured the maximum number of tries you've needed for the problematic GPU?  I wonder about shortening the sleep below 1 second and potentially increasing the maximum iterations.

Reply:
On my box it takes 1 try before it succeeds.  Will update with the if (rc) to prevent sleep on success for first iteration